### PR TITLE
use PrismicDocument in StoryPromo

### PR DESCRIPTION
**Pull request by jamesgorrie**
Follows the same pattern as #7372 by adding `prismicDocument` to articles.
We then pass this to the `StoryPromo`.

There's a lot more transformations added to the `StoryPromo` as we have a couple of hacks to work out position of an article within a series. Pretty nastly, but at least now it's contained. 

When adding a third type (probably events) I'\ll start pulling out commonlities and make sure theres testing across those.

#7363

### CODE REVIEW

File: common/model/editorial-series.ts [hunk 1]
```diff
@@ -1,4 +1,9 @@
-export type EditorialSeriesColour = 'turquoise' | 'red' | 'green' | 'purple';
+export type EditorialSeriesColour =
+  | 'teal'
```
Comment (alexwlchan) [inline common/model/editorial-series.ts:2]: Could this rely on the definition of ColorSelection so we don't have to remember to keep them both in sync?

File: content/webapp/services/prismic/series.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  RichTextField,
+  TimestampField,
+  PrismicDocument,
+  SelectField,
+} from '@prismicio/types';
+import { CommonPrismicData } from './types';
+
+export type SeriesPrismicDocument = PrismicDocument<
+  {
+    color: SelectField<'teal' | 'red' | 'green' | 'purple'>;
```
Comment (alexwlchan) [inline content/webapp/services/prismic/series.ts:14]: No turquoise?

**Review by alexwlchan — APPROVED** [review]
I don't have a deep understanding of this code yet, but it looks okay to me.

File: common/model/editorial-series.ts (hunk 1 above)
Comment (jamesgorrie) [reply common/model/editorial-series.ts:2]: It definitely could be defined in 1 place, I'll work with Gareth to see what these values should be.

File: content/webapp/services/prismic/series.ts (hunk 2 above)
Comment (jamesgorrie) [reply content/webapp/services/prismic/series.ts:14]: Nope, they've not had that choice previously. I'll tidy this up in my next PR.

**Pull request merged**